src/app/games/diam/DiamMove.ts

```diff
@@ -58,10 +59,14 @@ export class DiamMoveShift extends Move {
             return (this.start.x + 7) % 8;
         }
     }
-    public equals(other: DiamMoveShift): boolean {
-        if (this.start.equals(other.start) === false) return false;
-        if (this.moveDirection !== other.moveDirection) return false;
-        return true;
+    public equals(other: DiamMove): boolean {
+        if (other instanceof DiamMoveShift) {
+            if (this.start.equals(other.start) === false) return false;
+            if (this.moveDirection !== other.moveDirection) return false;
+            return true;
+        } else {
+            return false;
+        }
```

Comment:
if (other instanceof DiamMoveShift === false) return false;
if (this.start.equals(other.start) === false) return false;
return this.moveDirection === other.moveDirection;
?

Reply:
oui mais non, car alors faut caster `other` en `DiamMoveShift`, alors qu'avec le code actuel il *sait* que dans la branche then, `other` est un `DiamMoveShift` (il est pas assez intelligent pour savoir ça dans ton refactor, même si c'est aussi évident)

Reply:
ouais mias c'pas safe un `as` ! Si on ajoute un autre type de `DiamMove`, ton code il est pas cassé mais il throwera, le mien il sera cassé à la compil ! Rendons pas le code moins safe pour gagner 3 ligne :p